griptape/tokenizers/amazon_bedrock_tokenizer.py

```diff
@@ -0,0 +1,55 @@
+import os
+import json
+import boto3
+from attr import define, field, Factory
+from griptape.tokenizers import BaseTokenizer
+
+
+@define(frozen=True)
+class AmazonBedrockTokenizer(BaseTokenizer):
```

Comment:
@vasinov this only works with titan models since they're the only ones that return a token count. Should we rename this to `AmazonBedrockTitanTokenizer`?

Reply:
AI21 models served via Bedrock return the detail of each generated token, and the token count can be derived from that.

Reply:
Separated out `AmazonBedrockTokenizer` into `BedrockTitanTokenizer`, `BedrockJurassicTokenizer`, and `BedrockClaudeTokenizer`.

`BedrockTitanTokenizer` and `BedrockJurassicTokenizer` both use the token count from the `invoke_model`, though this is not ideal since it effectively doubles the cost. `BedrockJurassicTokenizer` can probably be swapped out for [ai21's tokenization API](https://docs.ai21.com/reference/tokenize-ref) but this requires an API key. Hopefully Amazon exposes proper tokenization APIs in the future.

`BedrockClaudeTokenizer` subclasses `AnthropicTokenizer` since we can use the same client-side tokenizer.